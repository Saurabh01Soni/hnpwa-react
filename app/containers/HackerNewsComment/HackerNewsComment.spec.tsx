import * as React from 'react';
import { shallow } from 'enzyme';

import { HackerNewsCommentInner } from './';

describe('<HackerNewsComment />', () => {
  it('should match snapshot when render default', () => {
    const wrapper = shallow(<HackerNewsCommentInner />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render comments and contents', () => {
    const contents = {
      content: '<p>How you think React with TypeScript?</p>',
      timeAgo: '1 years ago',
      user: 'taehwanno',
    };
    const comments = [1, 2, 3, 4];
    const wrapper = shallow(<HackerNewsCommentInner comments={comments} contents={contents} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have .HackerNewsComment__inner--hide when state.collapse === true', () => {
    const wrapper = shallow(<HackerNewsCommentInner />);
    wrapper.setState({ collapse: true });
    expect(wrapper.find('.HackerNewsComment__inner--hide').length).toBe(1);
  });

  it('should switch button shape by state.collapse', () => {
    const wrapper = shallow(<HackerNewsCommentInner />);
    expect(wrapper.state().collapse).toBe(false);
    expect(wrapper.find('button').children()).toMatchSnapshot();
    wrapper.setState({ collapse: true });
    expect(wrapper.find('button').children()).toMatchSnapshot();
  });

  it('should toggle state.collapse when simulate button click event', () => {
    const wrapper = shallow(<HackerNewsCommentInner />);
    expect(wrapper.state().collapse).toBe(false);
    wrapper.find('button').simulate('click');
    expect(wrapper.state().collapse).toBe(true);
    wrapper.find('button').simulate('click');
    expect(wrapper.state().collapse).toBe(false);
  });
});
