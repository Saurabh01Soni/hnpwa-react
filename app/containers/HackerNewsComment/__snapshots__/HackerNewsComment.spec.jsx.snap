// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HackerNewsComment /> should match snapshot when render default 1`] = `
<div
  className="HackerNewsComment"
>
  <div
    className="HackerNewsComment__head"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      [-]
    </button>
     
    <Link
      href="/user/undefined"
      replace={false}
      to="/user/undefined"
    />
     
    <span />
  </div>
  <div
    className="HackerNewsComment__inner"
  >
    <div
      className="HackerNewsComment__content"
      dangerouslySetInnerHTML={
        Object {
          "__html": undefined,
        }
      }
    />
  </div>
</div>
`;

exports[`<HackerNewsComment /> should render comments and contents 1`] = `
<div
  className="HackerNewsComment"
>
  <div
    className="HackerNewsComment__head"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      [-]
    </button>
     
    <Link
      href="/user/taehwanno"
      replace={false}
      to="/user/taehwanno"
    >
      taehwanno
    </Link>
     
    <span>
      1 years ago
    </span>
  </div>
  <div
    className="HackerNewsComment__inner"
  >
    <div
      className="HackerNewsComment__content"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>How you think React with TypeScript?</p>",
        }
      }
    />
    <Connect(HackerNewsCommentInner)
      commentId={1}
      key="1"
    />
    <Connect(HackerNewsCommentInner)
      commentId={2}
      key="2"
    />
    <Connect(HackerNewsCommentInner)
      commentId={3}
      key="3"
    />
    <Connect(HackerNewsCommentInner)
      commentId={4}
      key="4"
    />
  </div>
</div>
`;

exports[`<HackerNewsComment /> should switch button shape by state.collapse 1`] = `"[-]"`;

exports[`<HackerNewsComment /> should switch button shape by state.collapse 2`] = `"[+]"`;
